activesupport/lib/active_support/encrypted_file.rb

```diff
@@ -26,18 +26,21 @@ def self.generate_key
     end
 
 
-    attr_reader :content_path, :key_path, :env_key
+    attr_reader :content_path, :key_path, :env_key, :raise_if_missing_key
 
-    def initialize(content_path:, key_path:, env_key:)
+    def initialize(content_path:, key_path:, env_key:, raise_if_missing_key:)
       @content_path, @key_path = Pathname.new(content_path), Pathname.new(key_path)
       @env_key = env_key
+      @raise_if_missing_key = raise_if_missing_key
     end
 
     def key
       read_env_key || read_key_file || handle_missing_key
     end
 
     def read
+      return unless key
```

Comment:
Why do we need this?

Won't we also need to make sure `write` will raise without a key? And that `config` on Encrypted Configuration returns an empty hash?

Reply:
If there is no key check, it will decrypt with an empty key and `ActiveSupport::MessageEncryptor::InvalidMessage` will be raise, to prevent it. 

> Won't we also need to make sure write will raise without a key?  

Do we support write without key? I thought that it would be better just to prevent errors from being read. 
In the case of write, if the user incorrectly designates key, I think that encrypted files will be generated unintentionally. So, if do not specify key, do not have to support write, what do you think?

> And that config on Encrypted Configuration returns an empty hash? 

If the read result is empty, it is supposed to return an empty hash, so this is okay. 
https://github.com/rails/rails/blob/bc2e08034d4ee5e38536928ae0cb90210d27ddcf/activesupport/lib/active_support/encrypted_configuration.rb#L45

Reply:
> In the case of write, if the user incorrectly designates key, I think that encrypted files will be generated unintentionally. So, if do not specify key, do not have to support write, what do you think?

I'm not sure what you mean. I think it would make sense that `write` always raises if there's no key. Otherwise wouldn't users end up with a file that couldn't be *en*crypted? I guess it would still raise an error, we'd just rather have it be a KeyMissing one 😊